### DISCUSSION TITLE
Recent and Favourites for any game in subfolders are not showing images.

### DISCUSSION THREAD

**Discussion by dre10g** [OnionUI/Onion#1056] [Support] [answered]
### Onions OS Version

Onion V4.2.0-RC

### Miyoo Firmware Version

202306092122

### Provide a clear and concise description of the issue

Recent and Favourites for any game in subfolders are not showing images.
I have successfully setup my arcade section with 2 subfolders like this:
```
ARCADE
    >FBNEO
    >MAME
```

So they show up and boot up fine with their individual cores.
The images show up fine in the Games menu, but when I favourite a game from any of those 2 subfolders, it's image does not show in the the favourites list or recents list.

### Steps to Reproduce

Please provide detailed steps that will help us replicate and investigate your issue
Example:
Roms/ARCADE/myoogamelist.xml looks like this and works fine in Game list view..

```
  <game id="43837" source="ScreenScraper.fr">
    <path>./FBNEO/area88.zip</path>
    <name>Area 88</name>
    <image>./FBNEO/Imgs/unsquad.png</image>
  </game>
  <game id="43289" source="ScreenScraper.fr">
    <path>./MAME/ssrdrabd.zip</path>
    <name>Sunset Riders</name>
    <image>./MAME/Imgs/ssrdrabd.png</image>
  </game>
```

Roms/recentlist.json looks like this..

```
{"label":"Area 88","rompath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/./FBNEO/unsquad.zip","imgpath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/./FBNEO/Imgs/unsquad.png","launch":"/mnt/SDCARD/Emu/ARCADE/launch.sh","type":5}
{"label":"Sunset Riders","rompath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/./MAME/ssrdrabd.zip","imgpath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/./MAME/Imgs/ssrdrabd.png","launch":"/mnt/SDCARD/Emu/ARCADE/launch.sh","type":5} 
```

Emu/ARCADE/config.json looks like this..

```
{
	"label":	"Arcade",
	"icon":	"/mnt/SDCARD/Themes/TechNoir/icons/arcade.png",
	"launch":	"launch.sh",
	"rompath":	"../../Roms/ARCADE",
	"gamelist":	"../../Roms/ARCADE/miyoogamelist.xml",
	"useswap":	1,
	"shortname":	1,
	"hidebios":	1,
	"extlist":	"zip"
} 
```

### Provide any additional context or information that may help us investigate your issue

Setup individual cores according to this this wiki..
https://github.com/OnionUI/Onion/wiki/Combining-different-systems-or-cores

**Comment by Aemiii91** [comment]
You can't use multiple Imgs folders if you want it to show in recents and favorites.

**Comment by dre10g** [comment]
ok thanks

so i moved all the images into 1 Imgs folder under ARCADE and pointed each game's image path to that folder, so they all show up correctly in the regular games list view. But the images still dont show in recents or favourites.

Folder structure looks like this..

```
>Roms
  >ARCADE
    >FBNEO
    >Imgs
    >MAME
    miyoogamelist.xml
```

and miyoogamelist.xml looks like this..

```
 <game id="41785" source="ScreenScraper.fr">
    <path>./FBNEO/ikariram.zip</path>
    <name>Ikari Warriors (RAMBO)</name>
    <image>./Imgs/ikariram.png</image>
  </game>
  <game id="39911" source="ScreenScraper.fr">
    <path>./MAME/720.zip</path>
    <name>720°</name>
    <image>./Imgs/720.png</image>
  </game>
```

**Reply by Aemiii91** [reply]
Can you attach a copy of `Roms/favourite.json` and `Roms/recentlist.json` ?

**Comment by dre10g** [comment]
Sure.. attached (renamed json to txt)
[favourite.txt](https://github.com/OnionUI/Onion/files/12157038/favourite.txt)
[recentlist.txt](https://github.com/OnionUI/Onion/files/12157039/recentlist.txt)

**Reply by Aemiii91** [reply]
Try removing all the `./` path prefixes from the xml

**Reply by dre10g** [reply]
Example change this..

{"label":"Rolling Thunder 2","rompath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/./MAME2003PLUS/rthun2.zip","imgpath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/./Imgs/rthun2.png","launch":"/mnt/SDCARD/Emu/ARCADE/launch.sh","type":5}

to this?

{"label":"Rolling Thunder 2","rompath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/MAME2003PLUS/rthun2.zip","imgpath":"/mnt/SDCARD/Emu/ARCADE/../../Roms/ARCADE/Imgs/rthun2.png","launch":"/mnt/SDCARD/Emu/ARCADE/launch.sh","type":5}

**Reply by Aemiii91** [reply]
Yes, but also in the gamelist xml

**Comment by dre10g** [comment]
Also changed it in gamelist.. still not showing :( ..although it shows fine in the normal gamelist before and after i changed it.

` <game id="42838">
    <path>MAME2003PLUS/rthun2.zip</path>
    <name>Rolling Thunder 2</name>
    <image>Imgs/rthun2.png</image>
  </game>`

**Comment by Phanturian** [comment, marked as answer]
Someone correct me if I'm wrong (and I hope I am) but I'm pretty sure the miyoogamelist.xml doesn't support sub folders. My arcade build has over 1000 games split into 12 folders.  I would love the ability to rename games so they display alphabetically by game name, and not rom name, but I've been told it isn't possible time and time again.

I'm guessing this miyoogamelist.xml was generated on a computer and not the Miyoo Mini Plus? I just checked and there's no option to generate the xml within Onion OS that I can find. If all your Arcade games are in folders, I would just delete the miyoogamelist.xml and move all cover images into a Imgs folder inside the Arcade folder, alongside your game folders. I just tested this and the cover images show in favorites if there is no miyoogamelist.xml.

This is from the miyoogamelist guide provided here:

Limitations when using miyoogamelist.xml
• There is not yet a known way to use miyoogamelist with sub folders (it will display on the device only as
a single list/folder even if you scraped/generated a list from subfolders)

**Reply by Aemiii91** [reply]
You're probably right (unfortunately)

**Reply by dre10g** [reply]
The reason I have the arcade game split into folders is so i can assign the correct core to them by folder.
So under Roms, I have a folder ARCADE.. under that folder I have 3 folders names CPS, FBNEO & MAME2003PLUS.
This is because I assigned each folder to use a different core in the launch.sh file in the \Emu\ARCADE folder like this..

`# Select core based on name of containing folder
case `echo "$(basename "$homedir")" | awk '{print toupper($0)}'` in
    FBNEO)
        core="fbneo_libretro.so"
        ;;
    MAME2003PLUS)
        core="mame2003_plus_libretro.so"
        ;;
    CPS)
        core="fbalpha2012_libretro.so"
        ;;
    *)
        ;;
esac`

I created the miyoogamelist.xml on my pc by scraping each of theose individual folders with skraper, then cleaned up with the gamelist.xml to miyoogamelist.xml cleanup script to get rid of all the unwanted stuff, then copied and pasted thm all in 1 miyoogamelist.xml file located in \Roms\ARCADE. which also has the Imgs folder along side them like this..

Roms\ARCADE
----------------\CPS <--all my cps1/2/3 roms
----------------\FBNEO <--all my fbneo roms
----------------\Imgs <--all the image files named exactly like the rom filename
----------------\MAME2003PLUS <--all my mame2003plus roms

Using the miyoogamelist.xml the images show up perfectly in the Games menu (all in one list, which is how I want it)
But when I add any arcade game to favourites they do not show the images (neither in the Recents lists)

I tried what you suggested by removing the miyoogamelist.xml from the root of my ARCADE folder and refreshed, miyoo found all the games correctly but did not pickup the images from the ARCADE\Imgs folder (so in the Games list they didn't even show up). Which in turn, did not show up in the Recent\Favourites lists either.

Have I done something wrong to what you suggested?

**Reply by Aemiii91** [reply]
I think we're just gonna have to accept that box art on fav/recent doesn't work for arcade games 🤷‍♂️

**Comment by Aemiii91** [comment]
For good measure could you provide your `Roms/ARCADE/ARCADE_cache6.db` file?

**Reply by dre10g** [reply]
Sure.. attached
[ARCADE_cache6.zip](https://github.com/OnionUI/Onion/files/12166669/ARCADE_cache6.zip)

**Reply by dre10g** [reply]
umm.. did you change something in the latest rc update? I just updated now and check both recent and favourites and my arcade games are showing images now!

**Reply by Phanturian** [reply]
I'm glad you got it working. I can confirm that nothing changed between the Onion releases in terms of covers not showing in favorites. The covers for Arcade always showed in favorites for me. I did notice that in your miyoogamelist example above, the cover name did not match the rom name. This is fine if your miyoogamelist.xml sorts this info out, but when you remove the miyoogamelist like I suggested, the two names must match for the image to show.  You had this listed:

<game id="43837" source="ScreenScraper.fr">
    <path>./FBNEO/area88.zip</path>
    <name>Area 88</name>
    <image>./FBNEO/Imgs/unsquad.png</image>
  </game>

Without a miyoogamelist you would have to rename unsquad.png to area88.png and it would show properly. I commend you for your unique setup. My 12 folders within the ARCADE folder are all genres, and each game was tested and configured for MAME 2003 PLUS, MAME 2003 EXTREME, FBA 2012, or FBNeo. The difference between cores for certain games is insane. I have no use for a miyoogamelist.xml because my covers match the rom names and the miyoogamelist doesn't change the names of games inside folders no matter what you try.

**Reply by dre10g** [reply]
Cool thanks. Maybe I should go through and rename some of my image names to match the rom name and not use the XML. But for now it's working fine so I'm happy.

Curious though, which arcade games play better using mame 2003 extreme rather than the regular mame 2003 plus? Can you share your extreme games list?

**Reply by Phanturian** [reply]
I don't have my MAME Extreme games in a list, sorry. I would usually test with regular 2003 first, and then change the core to Extreme in GLO to check for differences. I prefer the regular MAME 2003+, as I feel the sound emulation is better. What Extreme does though is fix the controls on some games that are almost unplayable with MAME 2003+. Games like Afterburner, Space Harrier, Star Wars, Night Striker or any game that has yoke controls will snap the view back to the center of the screen if you let go of the d-pad. This makes for a very jerky and almost unplayable experience imo. With MAME Extreme, when you let go of the d-pad the view/reticle remains where you left it.

I prefer FBNeo the most, as the emulation is more accurate and the sound blows MAME 2003+ out of the water. More demanding games struggle with FBNeo on the Miyoo Mini (Plus), but a lot of games work amazing, especially older classic games. I find that FBA 2012 is especially good for vertical shooters like Armed Police Batrider, Batsugun, Gunbird, Guwange, etc. Basically if MAME struggles, and FBNeo also struggles, I'll check FBA 2012. The sound is very inferior, but many games were optimized to run on lower powered hardware using FBA.

Sorry if going off topic is against any rules here. Since were are in discussions, I hope it is acceptable.
